torch/csrc/distributed/c10d/symm_mem/CUDASymmetricMemoryOps.cu

```diff
@@ -1144,16 +1145,16 @@ at::Tensor stream_write_value32_(
   c10::cuda::CUDAGuard guard(input.device());
 
 #if !defined(USE_ROCM) && defined(PYTORCH_C10_DRIVER_API_SUPPORTED)
-  auto driver_api = c10::cuda::DriverAPI::get();
-  // According to the documentation of CUstreamWriteValue_flags,
-  // cuStreamWriteValue32 will provide a memory fence before the write, which
-  // has similar semantics to __threadfence_system() but is scoped to the
-  // stream rather than a CUDA thread.
-  C10_CUDA_DRIVER_CHECK(driver_api->cuStreamWriteValue32_(
+  static PFN_cuStreamWriteValue32_v2 pfn_cuStreamWriteValue32 = []() {
+    void *driver_ptr = c10::cuda::get_symbol("cuStreamWriteValue32");
```

Comment:
the ask was to change the way we are acquiring driver symbols throughout the codebase, not just fix a single callsite for driver API and wait for when the others will break.